### PR TITLE
[NFC] Move public symbols to top of file in e2ir.d

### PR DESCRIPTION
**Pull request by thewilsonator**
large diff. best viewed with whitespace off. No functional change only code motion

### CODE REVIEW

File: src/dmd/e2ir.d [hunk 1]
```diff
@@ -139,5584 +139,5859 @@ bool ISX64REF(IRState* irs, Expression exp)
...
+private __gshared StringTable!(Symbol*) *stringTab;
+
+elem *toElem(Expression e, IRState *irs)
+{
+    scope v = new ToElemVisitor(irs);
+    e.accept(v);
+    return v.result;
+}
+
+private:
```
Comment (thewilsonator) [inline src/dmd/e2ir.d:505]: Fold is here.

**Comment by ibuclaw** [comment]
5000 line diffs are never trivial. :-)

**Comment by RazvanN7** [comment]
I suggest splitting this PR into several commits as it is difficult to review it.

**Comment by thewilsonator** [comment]
there is no functional change in this PR, only functions moved around so that everything above line 505 is public (except `stringTab`) and everything below is private/

**Comment by RazvanN7** [comment]
Sorry, but this is unreviewable.

**Comment by CyberShadow** [comment]
> 5000 line diffs are never trivial. :-)

> Sorry, but this is unreviewable.

Perhaps this can help:

```console
~/work/extern/D/dmd » ( filter() { sed 's/^ *//'|sed 's/^private //'|sort } ; diff -u <(git show HEAD:src/dmd/e2ir.d|filter) <(git show HEAD^:src/dmd/e2ir.d|filter) )          
--- /proc/self/fd/11	2021-05-27 00:45:54.494592054 +0000                                                                                                           
+++ /proc/self/fd/12	2021-05-27 00:45:54.494592054 +0000
@@ -741,6 +741,11 @@
 
 
 
+
+
+
+
+
 )
 *
 *
@@ -5756,7 +5761,6 @@
 //printf("vs = %s, offset = %x, %p\n", vs.Sident, (int)vs.Soffset, vs);
 printHash();
 printHash();
-private:
 @property int REGSIZE() { return _tysize[TYnptr]; }
 *psym = stmp;
 *psym = symbol_genauto(Type_toCtype(tsarray2));
```

Thus, aside from:

- changing leading whitespace
- changing visibility
- adding some blank lines

This PR has no changes. LGTM :)

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**